Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,624 @@
+    The regular expression to validate against.
+    Must be a valid PCRE pattern, including delimiters (e.g. `/^...$/`).
+
+-   `message` (string, optional):
+    Custom error message or translation key. If not set, a localized default
+    message will be used.
+
+..  _extbase-validator-regularexpression-behavior:
+
+Validation behavior
+-------------------
+
+-   If the value does not match, an error is added.
+-   If the regex is invalid, an exception is thrown.
+-   The validator supports localized error messages via `LLL:EXT:...` syntax.
+
+..  _extbase-validator-regularexpression-example-basic:
+
+Example: username pattern
+-------------------------
+
+Validate that a value contains only alphanumeric characters:
+
+..  code-block:: php
+
+    use TYPO3\CMS\Extbase\Annotation\Validate;
+
+    class UserForm
+    {
+        #[Validate([
+            'validator' => 'RegularExpression',
+            'options' => [
+                'regularExpression' => '/^[a-z0-9]+$/i'
+            ]
+        ])]
+        public string $username = '';
+    }
+
+..  _extbase-validator-regularexpression-example-zip:
+
+Example: ZIP code
+-----------------
+
+Validate a 5-digit postal code with a custom error message:
+
+..  code-block:: php
+
+    use TYPO3\CMS\Extbase\Annotation\Validate;
+
+    class AddressForm
+    {
+        #[Validate([
+            'validator' => 'RegularExpression',
+            'options' => [
+                'regularExpression' => '/^\d{5}$/',
+                'message' => 'Bitte eine gültige Postleitzahl eingeben.'
+            ]
+        ])]
+        public string $postalCode = '';
+    }
+
+..  _extbase-validator-regularexpression-use-cases:
+
+Use cases
+---------
+
+-   Custom identifiers or slugs
+-   Postal/ZIP code validation
+-   Specific numeric or alphanumeric patterns
+
+..  _extbase-validator-regularexpression-important:
+
+Important
+---------
+
+Use this validator only for formats that are not supported by dedicated
+validators. Prefer these built-in validators when applicable:
+
+-   `EmailAddressValidator` – for email addresses
+-   `DateTimeValidator` – for dates
+-   `UrlValidator` – for URLs
+
+These are easier to configure, localized by default, and more robust.
+
+..  note::
+
+    - This is a valid list inside a note.
+    - Just make sure it's indented consistently.
+
+..  _extbase-validator-stringlength:
+
+StringLengthValidator
+=====================
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\StringLengthValidator`
+validates the length of a string.
+
+..  code-block:: php
+
+    #[Validate([
+        'validator' => 'StringLength',
+        'options' => ['minimum' => 5, 'maximum' => 50],
+    ])]
+    protected string $description;
+
+
+..  _extbase-validator-string:
+
+StringValidator
+===============
+
+The :php-short:`\TYPO3\CMS\Extbase\Validation\Validator\StringValidator`
+validates that a mixed variable is a string. Fails for array, numbers and bools.
```

Comment:
```suggestion
validates that a mixed variable is a string.
```

No need to mention what is not a string. ;-)

Reply:
This document is for integrators not only developers, also I think it is not that self-evident. In many contexts numbers and bools are treated like strings because they can be caseted to a string.